src/basic/calendarspec.c

```diff
@@ -471,6 +489,20 @@ static int prepend_component(const char **p, bool usec, CalendarComponent **c) {
 
                 if (repeat == 0)
                         return -ERANGE;
+        } else if (e[0] == '.' && e[1] == '.') {
+                e += 2;
+                r = parse_component_decimal(&e, usec, &range_end);
+                if (r < 0)
+                        return r;
+
+                if (value >= range_end)
+                        return -EINVAL;
+
+                for (i = value; i <= range_end; i += usec ? USEC_PER_SEC : 1) {
```

Comment:
not sure i understand the different increment here, regarding whether the `usec` bool is set or not...

Moreover I am a bit concerned that we might end up adding a ridiculously high number of chain elements (and thus allocate tons and tons of memory) in case µs specifications are given. Maybe we should generate a parse error in this case? Basically we really have three options here: do not accept ranges for µs specifications, accept it but enforce boundaries on the number of chain elements we generate, or interpret them semantically different, i.e. also use second increments in this case, so that the number of chain elements created here is automatically bound to 60...

I think this case also deserves covering in the test cases.